use of deleted/deprecated function: MPI_Handler_function

From https://travis-ci.org/pmodels/casper/jobs/613017969:
```
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I./include -I./src/hwloc/include/private/autogen -I./src/hwloc/include/hwloc/autogen -I./include -I./include -I./src/common/include -I./src/common/util -I./src/user/include -I./src/ghost/include -I/Users/travis/build/pmodels/casper/src/hwloc/include -I/Users/travis/build/pmodels/casper/src/hwloc/include -I./src/openpa/src -I/Users/travis/build/pmodels/casper/src/openpa/src -I/usr/include/libxml2 -std=c99 -Wno-unknown-attributes -MT src/user/errhan/errhandler_create.lo -MD -MP -MF src/user/errhan/.deps/errhandler_create.Tpo -c src/user/errhan/errhandler_create.c  -fno-common -DPIC -o src/user/errhan/.libs/errhandler_create.o
src/user/errhan/errhandler_create.c:12:27: error: unknown type name 'MPI_Handler_function'; did you mean 'MPI_User_function'?
int MPI_Errhandler_create(MPI_Handler_function * errhandler_fn, MPI_Errhandler * errhandler)
                          ^~~~~~~~~~~~~~~~~~~~
                          MPI_User_function
/usr/local/Cellar/open-mpi/4.0.2/include/mpi.h:448:15: note: 'MPI_User_function' declared here
typedef void (MPI_User_function)(void *, void *, int *, MPI_Datatype *);
              ^
1 error generated.
make[1]: *** [src/user/errhan/errhandler_create.lo] Error 1
```

--- a/src/user/errhan/errhandler_create.c
+++ b/src/user/errhan/errhandler_create.c
@@ -9,9 +9,13 @@
 #include <stdlib.h>
 #include <mpi.h>
 
+#ifndef OMPI_OMIT_MPI1_COMPAT_DECLS
+
 int MPI_Errhandler_create(MPI_Handler_function * errhandler_fn, MPI_Errhandler * errhandler)
 {
     /* Deprecated function, replaced by MPI_Comm_create_errhandler.
      * See overwritten content in MPI_Comm_create_errhandler. */
     return MPI_Comm_create_errhandler((MPI_Comm_errhandler_function *) errhandler_fn, errhandler);
 }
+
+#endif /* OMPI_OMIT_MPI1_COMPAT_DECLS */

--- a/src/user/errhan/errhandler_free.c
+++ b/src/user/errhan/errhandler_free.c
@@ -9,6 +9,8 @@
 #include <stdlib.h>
 #include "cspu.h"
 
+#ifndef OMPI_OMIT_MPI1_COMPAT_DECLS
+
 int MPI_Errhandler_free(MPI_Errhandler * errhandler)
 {
     int mpi_errno = MPI_SUCCESS;
@@ -27,3 +29,5 @@ int MPI_Errhandler_free(MPI_Errhandler * errhandler)
   fn_fail:
     goto fn_exit;
 }
+
+#endif /* OMPI_OMIT_MPI1_COMPAT_DECLS */

--- a/src/user/errhan/errhandler_get.c
+++ b/src/user/errhan/errhandler_get.c
@@ -7,9 +7,13 @@
 #include <stdlib.h>
 #include <mpi.h>
 
+#ifndef OMPI_OMIT_MPI1_COMPAT_DECLS
+
 int MPI_Errhandler_get(MPI_Comm comm, MPI_Errhandler * errhandler)
 {
     /* Deprecated function, replaced by MPI_Comm_get_errhandler.
      * See overwritten content in MPI_Comm_get_errhandler. */
     return MPI_Comm_get_errhandler(comm, errhandler);
 }
+
+#endif /* OMPI_OMIT_MPI1_COMPAT_DECLS */

--- a/src/user/errhan/errhandler_set.c
+++ b/src/user/errhan/errhandler_set.c
@@ -8,9 +8,13 @@
 #include <stdlib.h>
 #include <mpi.h>
 
+#ifndef OMPI_OMIT_MPI1_COMPAT_DECLS
+
 int MPI_Errhandler_set(MPI_Comm comm, MPI_Errhandler errhandler)
 {
     /* Deprecated function, replaced by MPI_Comm_set_errhandler.
      * See overwritten content in MPI_Comm_set_errhandler. */
     return MPI_Comm_set_errhandler(comm, errhandler);
 }
+
+#endif /* OMPI_OMIT_MPI1_COMPAT_DECLS */

